kit/ChildSession.cpp

```diff
@@ -1247,14 +1249,38 @@ bool ChildSession::clientVisibleArea(const StringVector& tokens)
     return true;
 }
 
-bool ChildSession::isTileInsideVisibleArea(const TileDesc& tile) const
+float ChildSession::getTilePriority(const std::chrono::steady_clock::time_point &now, const TileDesc &tile) const
 {
-    return tile.intersects( _clientVisibleArea );
-}
+    float score = tile.intersects(_clientVisibleArea) ? 2.0 : 1.0;
 
-bool ChildSession::isTileInsideVisibleArea(const TileCombined& tileCombined) const
-{
-    return tileCombined.toAABBox().intersects( _clientVisibleArea );
+    // most important to render things close to the cursor fast
+    if (tile.getPart() == _currentPart)
+    {
+        if (tile.intersects(_cursorPosition))
+            score *= 2.0;
+
+        // interacted with the keyboard/mouse recently ?
+        if (getInactivityMS(now) < 200 /* ms */) // typing etc.
+            score *= 2.0;
+
+        // pre-loading near the viewing area is also more important than far away
+        Util::Rectangle r = tile.toAABBox();
+        // grow in each direction
+        Util::Rectangle enlarged(r.getLeft() - r.getWidth(), r.getTop() - r.getHeight(),
+                           r.getRight() + r.getWidth(), r.getBottom() + r.getHeight());
+        if (enlarged.intersects(_clientVisibleArea))
```

Comment:
I think the 4 arg Util::Rectangle ctor is: int x, int y, int width, int height, while there is a ::create that takes x1, y1, x2, y2 which is what this looks like it wants.

Reply:
Grief; I was fooled by reading the rather odd variable initialization in the constructor somehow
    Rectangle(int x, int y, int width, int height)
        : _x1(x)
        , _y1(y)
        , _x2(x)
        , _y2(y)

Reply:
fixed; thanks =)